manager/src/main/java/com/redhat/service/smartevents/manager/workers/WorkManagerImpl.java

```diff
@@ -2,124 +2,87 @@
-                    w.getManagedResourceId(),
-                    w.getType()));
-        }
-        eventBus.requestAndForget(w.getType(), w);
-    }
-
-    @Transactional
-    protected void setModifiedAt(Work work) {
-        work.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
     }
 
     @Override
-    @Transactional
-    public boolean exists(Work work) {
-        return Objects.nonNull(workDAO.findById(work.getId()));
+    public void schedule(Bridge bridge) {
+        doSchedule(bridge.getId(), bridgeWorkerJob);
     }
 
     @Override
-    @Transactional
-    public void recordAttempt(Work work) {
-        if (!exists(work)) {
-            return;
-        }
-        // Work has been serialised by VertX at this point and has therefore lost all affinity with
-        // a JPA session. We therefore need to first retrieve the entity before updating it.
-        Work w = workDAO.findById(work.getId());
-        w.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
-        w.setAttempts(w.getAttempts() + 1);
-    }
-
-    @Override
-    @Transactional
-    public void complete(Work work) {
-        if (!exists(work)) {
-            return;
-        }
-        // Work has been serialised by VertX at this point and has therefore lost all affinity with
-        // a JPA session. We therefore need to delete it by Id and not the entity itself.
-        workDAO.deleteById(work.getId());
-    }
-
-    @SuppressWarnings("unused")
-    @Scheduled(cron = "{event-bridge.resources.workers.schedule}", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
-    protected void processWorkQueue() {
-        processWorkQueue(getWorkQueue());
+    public void schedule(Processor processor) {
+        doSchedule(processor.getId(), processorWorkerJob);
     }
 
-    @Transactional
-    protected List<Work> getWorkQueue() {
-        return workDAO.findByWorkerId(workerId);
-    }
-
-    @Transactional(Transactional.TxType.NEVER)
-    protected void processWorkQueue(List<Work> work) {
-        work.forEach(this::fireEvent);
-    }
+    // Schedule the first execution of a Job.
+    // We cannot use Quartz scheduler to invoke the job repeatedly as we store mutable state within the
+    // Trigger definition and Quartz does not support updating the Trigger JobDataMap after the Trigger
+    // has been created. Rescheduling of the Job is handled by AbstractWorker if either an execution
+    // failed or the Work remained incomplete.
+    protected void doSchedule(String id, JobDetail jobDetail) {
+        try {
+            TriggerKey key = new TriggerKey(id, MANAGED_RESOURCES_GROUP);
+            if (Objects.nonNull(quartz.getTrigger(key))) {
+                LOGGER.info(String.format("Work already exists for '%s'. Skipping.", id));
+            }
 
-    @SuppressWarnings("unused")
-    @Transactional
-    @Scheduled(every = "5m", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
-    void reconnectDroppedWorkers() {
```

Comment:
@manstis nice to see `reconnectDroppedWorkers` disappearing from our codebase :-)